radix-engine/src/engine/process.rs

```diff
@@ -360,24 +361,39 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
     > {
         #[cfg(not(feature = "alloc"))]
         let now = std::time::Instant::now();
-        re_info!(self, "Run started: function = {:?}", function);
+        re_info!(self, "Run started: snode_ref = {:?} function = {:?}", snode_ref, function);
 
         // Execution
         let output = match snode {
             SNodeState::Transaction(transaction_process) => {
                 transaction_process.main(self)
             }
             SNodeState::PackageStatic => {
-                Package::static_main(&function, args, self).map_err(RuntimeError::PackageError)
+                let arg = if args.len() > 1 {
+                    Err(RuntimeError::InvalidInvocation)
+                } else {
+                    args.into_iter().nth(0).ok_or(RuntimeError::InvalidInvocation)
+                }?;
+                Package::static_main(arg, self).map_err(RuntimeError::PackageError)
```

Comment:
Reason this was kept in is that it will all be refactored out in next PR when all methods have the same interface

Reply:
Maybe we need a better strategy w.r.t. intermediate changes, especially when we're certain they will be thrown away right after? I'm trying to avoid introducing a lot of "unreviewed" code in the code base, which is not a good practice.

Reply:
This is true, I can fix this then.